src/js/utils/MesosStateUtil.js

```diff
@@ -26,6 +26,19 @@ const MesosStateUtil = {
   },
 
   /**
+   * @param {{frameworks:array,completed_frameworks:array}} state
+   * @param {string} frameworkID
+   * @returns {{executors:Array, completed_executors:Array}|null} framework
+   */
+  getFramework(state, frameworkID) {
+    const {frameworks, completed_frameworks} = state;
+    return [].concat(frameworks, completed_frameworks).find(
+      function (framework) {
+        return framework != null && framework.id === frameworkID;
```

Comment:
Apparently `completed_frameworks` could be `null` but we could define the variables separately but I would vote to keep the null check anyway. What do you think? Should we change this?